### PR TITLE
Replace use of onRangeError

### PR DESCRIPTION
**Pull request by andralex**
This needs to be in in order to support non-pure onRangeError.

### CODE REVIEW

File: std/format.d [hunk 1]
```diff
@@ -6534,7 +6534,7 @@ immutable(Char)[] format(Char, Args...)(in Char[] fmt, Args args) if (isSomeChar
  */
 char[] sformat(Char, Args...)(char[] buf, in Char[] fmt, Args args)
 {
-    import core.exception : onRangeError;
+    import core.exception : onRangeError, RangeError;
```
Comment (atilaneves) [inline std/format.d]: Might as well delete the onRangeError import then.

Comment (quickfur) [inline std/format.d]: Agreed.

**Comment by quickfur** [comment]
Pushed the change manually, since @andralex seems too busy to respond to this. Will merge after autotester passes.

**Comment by andralex** [comment]
@quickfur much appreciated. Indeed I am on the road.

**Comment by quickfur** [comment]
Something strange is going on. The project tester seems to be indicating that vibe.d is failing to build? Don't understand the reason, though.

**Comment by ZombineDev** [comment]
@quickfur vibe.d was failing to build due to [linker errors](https://ci.dawg.eu/job/projects/2051/REPO=rejectedsoftware%2Fvibe.d/console). According to the [build history](https://ci.dawg.eu/job/projects/buildTimeTrend) the Project Tester started to fail consistently at [build 2006](https://ci.dawg.eu/job/projects/2006/) triggered by this [dmd pull request](https://github.com/dlang/dmd/pull/6666), though the auto-tester didn't manage to catch any linker problems with it. I wasn't able to find any hard rule for why some builds after [2006](https://ci.dawg.eu/job/projects/2006/) succeeded, except that some of them were targeting the stable branches of dmd.

**Pull request merged**